--- grub-orig/grub-2.02~rc1/grub-core/lib/syslinux_parse.c	2017-02-02 03:26:48.000000000 -0200
+++ grub-2.02~rc1/grub-core/lib/syslinux_parse.c	2017-03-08 03:50:18.213234880 -0300
@@ -962,7 +962,7 @@
 		      "linux_suffix=16; else linux_suffix= ; fi\n");
 	print_string ("  linux$linux_suffix ");
 	print_file (outbuf, menu, curentry->kernel_file, NULL);
-	print_string (" ");
+	print_string ("  $linux_extra ");
 	if (curentry->append)
 	  {
 	    err = print (outbuf, curentry->append, grub_strlen (curentry->append));
